client/blocks/plan-storage/index.jsx

```diff
@@ -34,7 +34,7 @@ class PlanStorage extends Component {
 			siteSlug,
 		} = this.props;
 
-		if ( jetpackSite ) {
+		if ( jetpackSite || ! sitePlan ) {
```

Comment:
Since we're not using the plan object for anything else, an alternative would be in connect to return the slug directly and check for truthyness of `sitePlanSlug` here. 
```js
const sitePlan = getSitePlan( state, siteId );
const sitePlanSlug = get( sitePlan, 'product_slug' );
```

In a future PR we should add an additional selector to cover this case